ts/utils/fetch.ts

```diff
@@ -35,8 +35,21 @@ function retryingFetch(
     maxRetries,
     exponentialBackoff
   );
+  const retryWithTransient429s: typeof retryLogic = (t, shouldAbort?) =>
```

Comment:
@chiadav 
To avoid code duplication what do you think about calling this function in both cases?
```javascript
const getTransientError = (
  task: TaskEither<Error | "transient", Response>,
  statusCode: Response["status"]
) =>
  task.chain(r =>
    fromEither(
      r.status === statusCode
        ? left<TransientError, never>(TransientError)
        : right<never, Response>(r)
    )
  );
```

Reply:
you probably want something like:

```ts
function retryLogicForTransientResponseError(p: (r: Response) => boolean): typeof retryLogic {
  return (t, shouldAbort?) =>
    retryLogic(
      // when the result of the task is a Response that satisfies
      // the predicate p, map it to a transient error
      t.chain(r =>
        fromEither(
          p(r)
            ? left<TransientError, never>(TransientError)
            : right<never, Response>(r)
        )
      ),
      shouldAbort
    );
}
```

then:

```ts
const retryWithTransient429s = retryLogicForTransientResponseError(_ => _.status === 429);
// ... 
const retryWithTransient404s = retryLogicForTransientResponseError(_ => _.status === 404);
```

Reply:
Using a predicate it is more flexible and "functional" 👌

Reply:
The function can be created in the io-ts-common.

Reply:
Function created in 9bb6cd0